packages/flutter/lib/src/rendering/debug.dart

```diff
@@ -105,6 +115,34 @@ List<String> debugDescribeTransform(Matrix4 transform) {
   return matrix;
 }
 
+void _debugDrawDoubleRect(Canvas canvas, Rect outerRect, Rect innerRect, Color color) {
+  final Path path = new Path()
+    ..addRect(outerRect)
+    ..addRect(innerRect);
+  final Paint paint = new Paint()
+    ..color = color;
+  canvas.drawPath(path, paint);
```

Comment:
The idea here is to punch a hole in the middle, not draw two overlapping rectangles. I may have gotten the winding rule we're using wrong, though, in which case it work work. I'll check that this actually works before landing.